homeassistant/components/accuweather/sensor.py

```diff
@@ -97,6 +97,7 @@ def __init__(
         self._device_class = None
         self._attrs = {ATTR_ATTRIBUTION: ATTRIBUTION}
         self.forecast_day = forecast_day
+        self._attr_state_class = self._description[ATTR_STATE_CLASS]
```

Comment:
If you do this:

```suggestion
        self._attr_state_class = self._description.get(ATTR_STATE_CLASS)
```

You can remove all the cases you set it to `None` in the constants file.

Reply:
Done